vspace/src/main/java/edu/asu/diging/vspace/core/file/impl/StorageEngine.java

```diff
@@ -21,38 +26,81 @@
-		File parent = new File(path + File.separator + directory);
-		if (!parent.exists()) {
-			parent.mkdir();
-		}
-		File file = new File(parent.getAbsolutePath() + File.separator + filename);
+    @Value("${uploads_path}")
+    private String path;
+
+
+    /* (non-Javadoc)
+    * @see edu.asu.diging.vspace.core.file.impl.IStorageEngine#storeFile(byte[], java.lang.String, java.lang.String)
+    */
+    @Override
+    public String storeFile(byte[] fileContent, String filename, String directory) throws FileStorageException {
+        
+        
+        File file = getFile(directory, filename);
         BufferedOutputStream stream;
         try {
             stream = new BufferedOutputStream(new FileOutputStream(file));
         } catch (FileNotFoundException e) {
-            throw new FileStorageException("Could not store file.", e);
+	    throw new FileStorageException("Could not store file.", e);
         }
         try {
             stream.write(fileContent);
             stream.close();
         } catch (IOException e) {
             throw new FileStorageException("Could not store file.", e);
         }
-        
+
         return directory;
-	}
-	
-	@Override
-	public byte[] getMediaContent(String directory, String filename) throws IOException {
-		File fileObject = new File(path + File.separator + directory + File.separator + filename);
+    }
+
+    @Override
+    public byte[] getImageContent(String directory, String filename) throws IOException {
+        File fileObject = getFile(directory, filename);       
+        return getFileContent(fileObject);
+    }
+
+    @Override
+    public boolean renameFile(String fileName, String newFileName, String directory) {             
+        File currentFile = getFile(directory, fileName);
+        File renamedFile = getFile(directory, newFileName);
+        return currentFile.renameTo(renamedFile);
+    }
+
+    @Override
+    public Resource downloadFile(String fileName, String directory) throws IOException {               
+        File fileObject = getFile(directory, fileName);
+        Path path = Paths.get(fileObject.getAbsolutePath());
+        return new ByteArrayResource(Files.readAllBytes(path));   
+    }
+
+    private byte[] getFileContent(File fileObject) throws IOException  {
+        URLConnection con = fileObject.toURI().toURL().openConnection();
+
+        InputStream input = con.getInputStream();
+
+        byte[] buffer = new byte[4096];
+
+        ByteArrayOutputStream byteOutput = new ByteArrayOutputStream();
+        BufferedOutputStream output = new BufferedOutputStream(byteOutput);
+
+        int n = -1;
+        while ((n = input.read(buffer)) != -1) {
+            output.write(buffer, 0, n);
+        }
+        input.close();
+        output.flush();
+        output.close();
+
+        byteOutput.flush();
+        byte[] bytes = byteOutput.toByteArray();
+        byteOutput.close();
+        return bytes;
+    }
```

Comment:
didn't notice until now, but this looks like almost the exact code as `getMediaContent`.